Lth/custom translations management translation file restructuring

lpandath opened the pull request:
<!-- Please make sure you have read the submission guidelines before posting an PR -->
<!-- https://github.com/tuwien-csd/damap-backend/blob/next/CONTRIBUTING.md#pullrequests -->

## Description

Adds a unified translation key structure and CSV mapping file to support translation customization through the admin interface.


#### What does this PR do?
Add translations.json containing all translation keys

Doc:
Add translations-mapping.csv with old_key, new_key, default_value columns 
[translations-mapping.csv](https://github.com/user-attachments/files/24097837/translations-mapping.csv)
[damap_DS_guide.pdf](https://github.com/user-attachments/files/24097845/damap_DS_guide.pdf)


### Checks

<!-- Adjust list as necessary -->

- [ ] Summary updated
- [ ] Version view updated
- [x] Documentation added
- [ ] Tests added
- [ ] E2e tests created
- [ ] Successfully ran e2e tests before merge

closes GH-497

lpandath committed:
feat: add unified translation structure files

lpandath committed:
add: add unified translation structure and CSV mapping file

DanceaVlad commented on CUSTOMISING.md:
```diff
@@ -136,6 +136,18 @@ existing keys (or simply use [assets/i18n/en.json](apps/damap-frontend/src/asset
+  - `default_value`: The default English translation value
+
+- **[translations-unified.json](apps/damap-frontend/src/assets/i18n/translations-unified.json)**: Contains all translation keys in a single unified JSON structure.
+
```
I think this is not the proper way to reference the changes.
If someone joins the team after the PR is merged, this will not give them good information about customizing the translations.

lpandath replied:
I agree. But this is how is done, as u see above. What would be your suggestion?

lpandath replied:
Probably would be best when the hole main feature is done to write an proper doc about it. I wrote only abou the part i did.

DanceaVlad replied:
Yes, I agree. CUSTOMISING.md should only be changed when the actual functionality is merged.
For now this does not add anything. There is already a **Translations** section, which directly contradicts these new descriptions.

lpandath replied:
i will add a proper doc about it when im done with the hole feature. Already started to doc for now locally.

DanceaVlad commented on apps/damap-frontend/src/assets/i18n/translations-mapping.csv:
I don't think this should be pushed to the repository, it is nice to have for the developers working on the PR but should be removed before pushing. Maybe the PR description should mention this.

lpandath replied:
sure, but for the future changes and u want to still want to analize the keys, would make sene to have it. You would probably need it later.

DanceaVlad replied:
Well after the PR is merged, this file will contain:
- old_key, which won't exist anymore
- new_key and default_value which is exactly what tranalations-unified.json contains.

Not only this, but if it is pushed to the repo it needs to be maintained along with the json file, which is a lot of extra work.
Also it can be uploaded to the PR description and it will stay there forever.

DanceaVlad commented on apps/damap-frontend/src/assets/i18n/translations.json:
I noticed that ~400 out of ~900 keys have remained the same. While I'm not doubting the result of the experiment, I still think they can be improved further. For example the ui.dashboard should be renamed to ui-dashboard, so the dots are reserved for parsing the JSON. I think this would help both developers and the admins.

lpandath replied:
Thank you but there is a logic behind, the hole idea as i planed for the hole keys. For the DS this was totally fine. 
[damap_DS_guide.pdf](https://github.com/user-attachments/files/23938689/damap_DS_guide.pdf)

DanceaVlad replied:
One more thing: I don't think the filename is appropriate, once the PR is merged and the old translation files are deleted, it won't make sense why they are _unified_. Maybe just `translations.json` or `default-translations.json`

DanceaVlad commented:
LGTM

DanceaVlad commented:
Dont forget to remove the txt file

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "access.noOwner": "Only owners of the DMP are allowed to change access rights.",
+    "access.no_rights": "Default setting. The user will not be able to see or interact with the DMP.",
+    "access.owner": "Owners can view, write, delete and assign permissions for DMPs. They are also capable of creating new owners - handle with care, new owners can remove old ones.",
+    "admin.banner": "Banner",
```
Minor: this key is not very verbos, should say something like section header or similar.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "admin.dialog.internal-storage.captions.backupLocation": "Backup Location",
+    "admin.dialog.internal-storage.captions.storageLocation": "Storage Location",
+    "admin.dialog.internal-storage.captions.url": "URL",
+    "admin.dialog.internal-storage.edit": "Edit internal storage",
```
I would switch dialog and internal-storage/banner keys here, to make it easier to navigate the keys.

Should also be done for similar instances.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "admin.table.header.storageLocation": "Storage Location",
+    "admin.table.header.title": "Title",
+    "admin.table.header.url": "URL",
+    "admin.table.no-selected": "No internal storage selected",
```
There is an internal-storage missing here, this probbaly wont be the only table in the admin section.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "ui.button.preview.content": "Show preview",
+    "ui.button.preview.tooltipDisabled": "Please save the DMP before being able to preview it.",
+    "ui.button.save": "Save changes",
+    "ui.button.version": "Save new version",
```
UI is  abad key, since everything is the UI - in this case, it should be replaced with something like action bar, choose fitting replacements for the other occurences.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "dmp.steps.data.specify.answer.kind.reuse.none": "No data will be reused.",
+    "dmp.steps.data.specify.answer.kind.specify": "Specify datasets.",
+    "dmp.steps.data.specify.answer.kind.unknown": "I don't know yet.",
+    "dmp.steps.data.specify.button.add.doi": "Add dataset",
```
This key will not be clear

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,891 @@
+    "dmp.steps.data.specify.answer.kind.unknown": "I don't know yet.",
+    "dmp.steps.data.specify.button.add.doi": "Add dataset",
+    "dmp.steps.data.specify.button.add.manual": "Add dataset manually",
+    "dmp.steps.data.specify.dialog.add": "Add dataset",
```
This key will not be clear

ValentinFutterer reviewed (commented):
Disclaimer, this is not a complete review, I only outlined some issues that immediately popped up on a first viewing.

I think you created this file with a script, which is fine at first, but it needs a manual review and polishing. The old keys were not generated with admin interaction in mind and were badly named and or relied on the fact that the files where split and you would view the code together with the files. This creates some key combinations that will cause confusion and require reworking on our part - which will be expensive with the new system, so I would rather avoid it as far as possible.

lpandath commented:
> Disclaimer, this is not a complete review, I only outlined some issues that immediately popped up on a first viewing.
> 
> I think you created this file with a script, which is fine at first, but it needs a manual review and polishing. The old keys were not generated with admin interaction in mind and were badly named and or relied on the fact that the files where split and you would view the code together with the files. This creates some key combinations that will cause confusion and require reworking on our part - which will be expensive with the new system, so I would rather avoid it as far as possible.

Thanks for the heads-up. I did a full manual pass after the script:

Access: clarified the short key (access.instructions.assignableUsers).

Admin: reorganized to feature-first (admin.banner.*, admin.internalStorage.*), grouped dialogs/tables/actions, and filled in missing internal-storage table bits.

Data specify: renamed the ambiguous answer/button keys to clearer ones.

UI: replaced the generic ui.* with scoped prefixes (dashboard.*, layout.*, plans.*, buttons.*, dialogs.common.*, forms.*, consent.*, landing.*, preview.*, treeselect.*, upload.*) so keys make sense on their own.

Mapping: regenerated the CSV with all old → new keys and defaults.

updated files:
[translations-mapping-1.csv](https://github.com/user-attachments/files/24185504/translations-mapping-1.csv)
[damap_loc_guide.pdf](https://github.com/user-attachments/files/24185524/damap_loc_guide.pdf)

DanceaVlad commented:
I went through the translation keys and I think many of them need further refinement. Here is what I found:

- since you no longer have access to the individual translations, the admin lacks context for where each key is found
- since there is no real limit to how long the keys can be, I think including more (or too much) context is better than not enough. For example:
  - `admin.banner` can be changed to `admin-page.banner` and `` can be changed to ``
- some elements I think are not self explainable to a non-tech-savy person. For example:
  - `banner.dialog` can be changed to `banner.pop-up`
- some keys are supposed to have another overarching key: For example:
  - ```json
      "buttons.actions": "Actions",
      "buttons.discard": "Discard changes",
      "buttons.export": "Export document",
      "buttons.preview.content": "Show preview",
      "buttons.preview.tooltipDisabled": "Please save the DMP before being able to preview it.",
      "buttons.save": "Save changes",
      "buttons.version": "Save new version",

  - these should be something like:
  - ```json
      "actions-bar.title": "",
      "actions-bar.buttons.discard": "",
      "actions-bar.buttons.export": "",
      "actions-bar.buttons.preview.content": "",
      "actions-bar.buttons.preview.hover-message-when-disabled": "", // change tooltip, i thing it is too techy
      "actions-bar.buttons.save": "",
      "actions-bar.buttons.version": "",
- some keys have redundant overarching keys. For example:
  - `dmp.license.wizard.info` can be changed to `license.wizard.info`
- some keys use `.` to denote word spaces, instead of going 'into' the feature deeper. For example:
  - `dmp.license.wizard.info` denotes that in the `dmp` there is a `license` section and in that section there is a `wizard` section, when if fact it is a `license wizard`. So I think it can be changed to `license-wizard.info`
- a few keys have absolutely no context to them. For example:
  - `dmp.no` and `dmp.or`
- some keys use words that are too general. For example:
  - many elements from the `layout` overarching key could be changed to something more specific. Group some of the `layout.x` keys into `sidebar.`. Even further you can create submodules in it, like `sidebar.footer.gdpr` and `sidebar.navigation-buttons.home`.
- some keys can be moved to a completely different place. For example:
  - `layout.menu.titleDashboard` is text that only appears when you are on the first step, so it isn't technically part of the layout
- i think it would be worth making a script that checks if the keys are actually used in the project and removes them from the json file. I made one quickly that just checks and got this `Found and kept: 692
Removed (unused): 189`

lpandath commented:
> I went through the translation keys and I think many of them need further refinement. Here is what I found:
> 
>     * since you no longer have access to the individual translations, the admin lacks context for where each key is found
> 
>     * since there is no real limit to how long the keys can be, I think including more (or too much) context is better than not enough. For example:
>       
>       * `admin.banner` can be changed to `admin-page.banner` and `can be changed to`
> 
>     * some elements I think are not self explainable to a non-tech-savy person. For example:
>       
>       * `banner.dialog` can be changed to `banner.pop-up`
> 
>     * some keys are supposed to have another overarching key: For example:
>       
>       * ```json
>           "buttons.actions": "Actions",
>           "buttons.discard": "Discard changes",
>           "buttons.export": "Export document",
>           "buttons.preview.content": "Show preview",
>           "buttons.preview.tooltipDisabled": "Please save the DMP before being able to preview it.",
>           "buttons.save": "Save changes",
>           "buttons.version": "Save new version",
>         ```
>       * these should be something like:
>       * ```json
>           "actions-bar.title": "",
>           "actions-bar.buttons.discard": "",
>           "actions-bar.buttons.export": "",
>           "actions-bar.buttons.preview.content": "",
>           "actions-bar.buttons.preview.hover-message-when-disabled": "", // change tooltip, i thing it is too techy
>           "actions-bar.buttons.save": "",
>           "actions-bar.buttons.version": "",
>         ```
> 
>     * some keys have redundant overarching keys. For example:
>       
>       * `dmp.license.wizard.info` can be changed to `license.wizard.info`
> 
>     * some keys use `.` to denote word spaces, instead of going 'into' the feature deeper. For example:
>       
>       * `dmp.license.wizard.info` denotes that in the `dmp` there is a `license` section and in that section there is a `wizard` section, when if fact it is a `license wizard`. So I think it can be changed to `license-wizard.info`
> 
>     * a few keys have absolutely no context to them. For example:
>       
>       * `dmp.no` and `dmp.or`
> 
>     * some keys use words that are too general. For example:
>       
>       * many elements from the `layout` overarching key could be changed to something more specific. Group some of the `layout.x` keys into `sidebar.`. Even further you can create submodules in it, like `sidebar.footer.gdpr` and `sidebar.navigation-buttons.home`.
> 
>     * some keys can be moved to a completely different place. For example:
>       
>       * `layout.menu.titleDashboard` is text that only appears when you are on the first step, so it isn't technically part of the layout
> 
>     * i think it would be worth making a script that checks if the keys are actually used in the project and removes them from the json file. I made one quickly that just checks and got this `Found and kept: 692 Removed (unused): 189`

As mention on the description, I kept the keys page-scoped to land the structure first including on the side the guide so that is very clear on the keys applied.
Def i can go deeper with the context if different suggested (... admin-page.banner, banner.pop-up, actions-bar.buttons.*, license-wizard.*, more specific sidebar. ...). 

noted: also agree we should prune unused keys;

Lets discuss in our meeting today, before ill apply those changes.

lpandath commented:
Total keys in en.json union: 881 translation keys
Used in code and including .ts files (we have some keys there aswell,  e.q question forms: 665
Unused: 216 translation keys - got removed. 

[translations-mapping.csv](https://github.com/user-attachments/files/24200373/translations-mapping.csv)
[damap_loc_guide.pdf](https://github.com/user-attachments/files/24200145/damap_loc_guide.pdf)

lpandath commented:
> Total keys in en.json union: 881 translation keys Used in code and including .ts files (we have some keys there aswell, e.q question forms: 665 Unused: 216 translation keys - got removed.
> 
> [translations-mapping.csv](https://github.com/user-attachments/files/24200373/translations-mapping.csv) [damap_loc_guide.pdf](https://github.com/user-attachments/files/24200145/damap_loc_guide.pdf)



> Total keys in en.json union: 881 translation keys Used in code and including .ts files (we have some keys there aswell, e.q question forms: 665 Unused: 216 translation keys - got removed.
> 
> [translations-mapping.csv](https://github.com/user-attachments/files/24200373/translations-mapping.csv) [damap_loc_guide.pdf](https://github.com/user-attachments/files/24200145/damap_loc_guide.pdf)

sorry, miscalculated by hand. In the translation JSON file there are 191 keys. 
So unused: 191 translation keys - got removed.

lpandath commented:
hi, due to the some number of mixed keys or even unused keys, we have manually reviewed them all. 

@DanceaVlad thanks for double checking the dashboard page keys.

Here are my updates:
```
  dmp.steps                      256
  license-wizard.license         86
  dmp.versions                   66
  admin.internalStorage          37
  http.error                     32
  plans.table                    17
  admin.banner                   15
  dashboard.tool                 13
  info.ethical                   13
  license-wizard.question        13
  dashboard.cards                9
  dashboard.instructions         9
  sidebar.footer                 9
  validation.error               9
  dashboard.dmp                  7
  license-wizard.answer          7
  http.success                   6
  actions-bar.buttons            5
  consent-popup.personalData     5
  version-save.dialog            4
```

here is our shared file: 
[translations.json (1).xlsx](https://github.com/user-attachments/files/24255562/translations.json.1.xlsx)

here is the updated guide in the end: 
[damap_loc-guide_latest.pdf](https://github.com/user-attachments/files/24255590/damap_loc-guide_latest.pdf)

lpandath committed:
add translations restructure and drop unused tree-select keys

lpandath commented:
Admin section for updating translations - new keys:

```
"admin.appTranslations.title": "Update application translations",
"admin.appTranslations.subtitle": "Customize text per language without code changes.",
"admin.appTranslations.languageLabel": "Language",
"admin.appTranslations.addLanguage": "Add language",
"admin.appTranslations.addLanguageHint": "Activates the language and copies default texts.",
"admin.appTranslations.languageCode": "Language code",
"admin.appTranslations.languageCodeHint": "Use a two-letter ISO 639-1 code (e.g. en, de, fr).",
"admin.appTranslations.languageCodeRequired": "Language code is required.",
"admin.appTranslations.languageCodePattern": "Use a two-letter language code (a-z).",
"admin.appTranslations.removeLanguage": "Remove language",
"admin.appTranslations.removeLanguageWarning": "WARNING: Removing this language will permanently delete ALL custom translations for this language. This action cannot be undone.",
"admin.appTranslations.searchPlaceholder": "Search by key or text",
"admin.appTranslations.searchPlaceholderExample": "e.g. admin.banner",
"admin.appTranslations.sectionFilter": "Section",
"admin.appTranslations.sectionAll": "All sections",
"admin.appTranslations.valueFilter": "Value type",
"admin.appTranslations.valueFilterAll": "Default and custom",
"admin.appTranslations.valueFilterCustom": "Custom only",
"admin.appTranslations.valueFilterDefault": "Default only",
"admin.appTranslations.keyColumn": "Key",
"admin.appTranslations.valueColumn": "Value",
"admin.appTranslations.currentValueFor": "Current value for",
"admin.appTranslations.customValueFor": "Custom value for",
"admin.appTranslations.save": "Save",
"admin.appTranslations.loading": "Loading translations...",
"admin.appTranslations.empty": "No translations match your filters.",
"admin.appTranslations.customBadge": "Custom",
"admin.appTranslations.defaultBadge": "Default",
"admin.appTranslations.cancel": "Cancel",
"admin.appTranslations.add": "Add",
"admin.appTranslations.languages.exists": "Language already exists.",
"admin.appTranslations.languages.cannotDeleteDefault": "The default language cannot be removed."
```

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "sidebar.footer.data-protection-declaration.link": "#",
+    "sidebar.footer.collapsed.hint": "Expand the menu to access legal information",
+    "info.damap.title.text": "About DAMAP",
+    "info.damap.title.link": "https://www.damap.org",
```
Follwing the established conventions, this key should be named something like dashboard.info.title.text

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dashboard.tool.bulletpoints.benefits.text": "Saves you work by",
+    "dashboard.tool.bulletpoints.benefits.three": "Suggesting answers that you can either comply with or adjust to your needs",
+    "dashboard.tool.bulletpoints.benefits.two": "Providing wizards, guidance, and item lists to choose from",
+    "dashboard.tool.bulletpoints.dmp.sepg.link": "https://www.scienceeurope.org/our-resources/practical-guide-to-the-international-alignment-of-research-data-management",
```
As mentioned above, tool is the wrong key here imo. The section should be named about (or something similar) to more clearly convey what it is

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dashboard.tool.bulletpoints.madmp.rda.link": "https://doi.org/10.15497/rda00039",
+    "dashboard.tool.bulletpoints.madmp.rda.text": "RDA recommendation on machine actionable DMPs",
+    "dashboard.tool.bulletpoints.madmp.suffix": ".",
+    "dashboard.tool.bulletpoints.supports": "Delivers DMPs for FWF and Horizon Europe projects in the funder-specific formats",
```
As a general comment about the dashboard keys: I do believe the dashboard parent key should be removed from most of them, since in the new UI, most of them are under a different route than dashboard. 

Stuff like this is historic debt from the old UI and has to be addressed.

lpandath replied:
@ValentinFutterer why remove? we already split it this way. I think if we again change the structure, then this will effect everything! Could we not stick to a plan? What do you think?

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "admin.internalStorage.translation.sectionTitle": "Translations for: ",
+    "admin.internalStorage.translation.table.header.languageCode": "Language Code",
+    "admin.pageTitle": "Admin page",
+    "app.title": "DAMAP",
```
I dont think this key is used, but correct me if I am wrong.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "common.choice.no": "No",
+    "common.choice.or": "or",
+    "common.dialog.button.cancel": "Cancel",
+    "common.languages": "Languages",
```
From this, icannot tell where it is used. Or is it truly common and used everywhere in the application?

ValentinFutterer replied:
Are you sure they are not used somewhere? They would have to have come from somewhereand I dont think you hallucinated them.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "delete-dialog.message.dmp": "You are about to delete this DMP. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dialog.message.storage": "You are about to delete this storage. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dialog.message.storageTranslation": "You are about to delete this storage translation. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dialog.title": "Please note:",
```
Minor, maybe make it clear that its a dmp delete dialog

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "delete-dialog.message.storageTranslation": "You are about to delete this storage translation. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dialog.title": "Please note:",
+    "dmp-list.load.error": "DMPs were not loaded.",
+    "dmp.contenttypes": "Content types",
```
This is misnamed, I think its somehwere from repositories?

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "delete-dialog.title": "Please note:",
+    "dmp-list.load.error": "DMPs were not loaded.",
+    "dmp.contenttypes": "Content types",
+    "dmp.dialog.title": "Save new version",
```
Make clear its from the version dialog

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.contenttypes": "Content types",
+    "dmp.dialog.title": "Save new version",
+    "dmp.fair": "FAIR",
+    "dmp.metadatastandards": "Metadata standards",
```
They dont belonmg here, are they also from repositories

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+        "title": "Data Management Costs Link",
+        "text": " to find out more about possible data management costs that may also be eligible for funding."
+      }
+    ],
```
We wanted to flatten everything out, but here is  alist

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.costs.question.title": "Name",
+    "dmp.steps.costs.question.type": "Cost Type",
+    "dmp.steps.costs.question.value": "Estimated cost",
+    "dmp.steps.costs.rightLabel": "Data management costs",
```
"other" is the wrong key here
"title" should be "name"
"rightlabel": the label isnt even to the right and its not a label, its basically a section title

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.data.reuse.answer.targetAudience.others": "Others:",
+    "dmp.steps.data.reuse.question.targetAudience": "Who is the target audience and who could be interested in reusing the data and why?",
+    "dmp.steps.data.reuse.question.tools": "Indicate whether potential users need specific tools or software to access and (re-)use the data.",
+    "dmp.steps.data.reuse.rightLabel": "Data reuse planning",
```
Every rightlabel should be looked at, so from now i wont mention them anymore

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.data.specify.table.header.size": "Size",
+    "dmp.steps.data.specify.table.header.source": "Source",
+    "dmp.steps.data.specify.table.heading.new": "New data",
+    "dmp.steps.data.specify.table.heading.reused": "Existing data",
```
Those two are not table headings, they are the side buttoins to switch view

ValentinFutterer replied:
This is still in the file.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.data.specify.tab.access": "Access rights",
+    "dmp.steps.data.specify.tab.cancel": "Cancel",
+    "dmp.steps.data.specify.tab.storage": "Storage",
+    "dmp.steps.data.specify.tab.upload": "Upload sample data",
```
Why are they all under "tab"

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.data.specify.dialog.source": "Source",
+    "dmp.steps.data.specify.dialog.technicalResource": "Technical Resource",
+    "dmp.steps.data.specify.dialog.title": "Dataset title",
+    "dmp.steps.data.specify.dialog.title.create": "Add dataset",
```
Those two can be confusing

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.data.specify.dialog.title": "Dataset title",
+    "dmp.steps.data.specify.dialog.title.create": "Add dataset",
+    "dmp.steps.data.specify.dialog.type": "Estimated type",
+    "dmp.steps.data.specify.doi.not.found": "No dataset found.",
```
Minor: no dot needed between "not.found"

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.people.confirm.dialog.button.cancel": "Cancel",
+    "dmp.steps.people.confirm.dialog.button.confirm": "Confirm",
+    "dmp.steps.people.confirm.dialog.heading": "Remove contributor",
+    "dmp.steps.people.confirm.dialog.question": "Are you sure you want to remove this contributor?\n  They are responsible for the deletion of the following datasets:",
```
There is no dialog in step 2, are those translations even used? Please check and rename if they are used.

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.steps.storage.external.add": "Add other storage location",
+    "dmp.steps.storage.external.question.backup.frequency": "Backup Frequency",
+    "dmp.steps.storage.external.question.backup.location": "Backup Location",
+    "dmp.steps.storage.external.question.info": "Please explain why institutional storage will not be used",
```
Minor: info is not very verbose and doesnt describe the text well

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "dmp.versions.view.storage.heading": "Storage",
+    "dmp.versions.view.storage.location": "Storage location",
+    "dmp.versions.view.storage.url": "URL",
+    "dmp.yes": "Yes",
```
No idea what or where that is 😅

lpandath replied:
"yes" not used, removed

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "export-dialog.message": "Please note that you are not done yet. \n  The exported document may still require fine-tuning.",
+    "export-dialog.title": "The file download has started.",
+    "form-status.changes.saved": "All changes have been saved.",
+    "form-status.changes.unsaved": "You have unsaved changes.",
```
Isnt this part of the actions-bar?

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "help.ethical.ethicalIssues": "For example: Your research involves human embryos, foetuses, cells or tissues, animals, activities that may adversely affect the environment or the health and safety of involved persons, artificial intelligence, activities outside the EU, or there are dual use or misuse aspects that must be addressed.",
+    "help.ethical.humanParticipants": "The participation may include activities such as interviews, prototype testing, observations, surveys, etc.",
+    "help.legal.personalDataCompliance": "Make sure you follow your institutions guidance on informed consent.",
+    "help.legal.sensitiveDataSecurity": "Referring to physical security, network security, and security of computer systems and files.",
```
Those should be under the corresponding "dmp.steps" sections

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,711 @@
+    "info.ethical.humanParticipants.researchEthics.text": "center for responsible research practices",
+    "info.ethical.humanParticipants.researchEthicsCoordination.link": "https://www.tuwien.at/fileadmin/Assets/forschung/Forschungsethik/Guidance_Document_Informed_Consent.pdf",
+    "info.ethical.humanParticipants.researchEthicsCoordination.text": "Example Informed Consent Guide",
+    "info.repositories.retention": "Set the time for how long your data should remain in the repository",
```
Like above

ValentinFutterer replied:
First two were transferred, but not deleted in their original spot, so they are duplicates right now.

ValentinFutterer reviewed (commented):
I have taken (relatively) quick look and its immediately way more organized than last time I saw it, like the changes a lot :)
In the comments, I highlighted smaller issues that I found, if  I misunderstood soemthing, feel free to correct in the comments.

One other thing i wanted to comment on, was the use of label and heading, since it confused me a little. I think label is the title in the stepper component and heading is the title of the buttons to switch views inside the stepper? But heading and label are also used in other key types, so some differentiation beetwen them migth be nice.

Dashboard keys also need to be split into the different URL parts, I think a mistake happenend here, since this part of the file seems almost untouched.

lpandath committed:
refactor: reorganize translation keys to dmp.steps structure

- Move info.* keys to corresponding dmp.steps.* sections
- Move help.* keys to dmp.steps.*.help.* sections
- Rename rightLabel to sectionTitle throughout
- Rename tab.* keys to appropriate namespaces (view.*, button.*, dialog.*)
- Move form-status.* to actions-bar.status.*
- Fix naming conventions (not.found -> notFound, etc.)
- Remove unused dmp.yes key

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,734 @@
+    "sidebar.footer.data-protection-declaration.link": "#",
+    "sidebar.footer.collapsed.hint": "Expand the menu to access legal information",
+    "dashboard.home.section-intro": "Welcome to DAMAP, a service that helps you create a Data Management Plan (DMP) for your project.",
+    "dashboard.cards.instructions-info": "You can create a new DMP by clicking the 'Create New DMP' butto",
```
typo

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,734 @@
+    "dashboard.instructions.update.heading": "Regularly update your DMP.",
+    "dashboard.tool.heading": "About DAMAP",
+    "dashboard.tool.description": "DAMAP guides you through creating a DMP using pre-filled information and customizable templates. It aligns with Science Europe guidelines and national funder requirements, saving you time and effort.",
+    "dashboard.tool.link": "https://www.damap.org",
```
They all belong under "info.how-to-create" not "dashboard". Please do the same you also did with "info.about".

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,734 @@
+    "delete-dmp-dialog.message.dmp": "You are about to delete this DMP. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dmp-dialog.message.storage": "You are about to delete this storage. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dmp-dialog.message.storageTranslation": "You are about to delete this storage translation. This action cannot be undone. Are you sure you want to proceed?",
+    "delete-dmp-dialog.title": "Please note:",
```
Sorry for my last comment, I didnt see the "message.storage". Seems like these are shared keys, between the action-bar and part of the admin UI? We should unravel them to not have these weird key-conglomerations, what do you think?

E.g. version save dialog has its own keys. On a similar note, they all have "Save", "Cancel" and similar words. Maybe we should really have common keys for those, whats your opinion on that.

lpandath replied:
Lets use one generic delete dialog namespace - delete.dialog.*. Its shared  and we can still have specific messages inside it like delete.dialog.message.dmp, delete.dialog.message.storage, etc.”

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,734 @@
+    "dmp.steps.costs.question.exist.label": "Select the statement which best describes the situation in your project with respect to costs of data management.",
+    "dmp.steps.costs.question.exist.no": "There are no costs dedicated to data management.",
+    "dmp.steps.costs.question.exist.yes": "As outlined below, there are costs dedicated to data management.",
+    "dmp.steps.costs.question.customType": "Please specify the cost type",
```
"customType" seems a little out of place, why not something like "specify-type".

ValentinFutterer commented on apps/damap-frontend/src/assets/i18n/translations.json:
```diff
@@ -0,0 +1,734 @@
+    "dmp.steps.data.specify.question.nodata": "Please explain why you won't create or analyse any research data.",
+    "dmp.steps.data.specify.question.reused": "What kinds of research data will you reuse?",
+    "dmp.steps.data.specify.view.access": "Access rights",
+    "dmp.steps.data.specify.view.storage": "Storage",
```
Arent those two from step 5? I dont see them anywhere in step 3.

ValentinFutterer reviewed (commented):
Only a few fixes left :)
I have also mentioned this in a comment, but a "common" key for things like clear, cancel, yes, no, etc would maybe be benefecial
Whats your opinion?

lpandath commented:
@ValentinFutterer   common.* keys wer not referenced in the old implementation. The old UI uses top‑level yes/no and dialog‑specific keys like dialog.delete.cancel, not common.choice.* or common.dialog.button.cancel.

ValentinFutterer commented:
> @ValentinFutterer common.* keys wer not referenced in the old implementation. The old UI uses top‑level yes/no and dialog‑specific keys like dialog.delete.cancel, not common.choice.* or common.dialog.button.cancel.

Yes I know that they didnt appear in the old implementation. But do we want a generic toplevel key is my question. Of course we can have the generic words duplicated in all sections, but wouldnt it be easier to have the generic keys? This would also help when replacing all the old transaltions - since we dont have a mapping from old keys to new, we have to go by value - and the duplicates you will need to apply manually. If you create generic keys, it will save a lot of work there.

lpandath commented:
> > @ValentinFutterer common.* keys wer not referenced in the old implementation. The old UI uses top‑level yes/no and dialog‑specific keys like dialog.delete.cancel, not common.choice.* or common.dialog.button.cancel.
> 
> Yes I know that they didnt appear in the old implementation. But do we want a generic toplevel key is my question. Of course we can have the generic words duplicated in all sections, but wouldnt it be easier to have the generic keys? This would also help when replacing all the old transaltions - since we dont have a mapping from old keys to new, we have to go by value - and the duplicates you will need to apply manually. If you create generic keys, it will save a lot of work there.

yes, i suggested that and added it.

BlueTechie committed:
Implement feedback + new keys

BlueTechie commented:
Hi, I added a commit that addresses the remaining feedback and also adds new translation keys introduced by these PRs, which will soon be merged:
- https://github.com/damap-org/damap-frontend/pull/486/files 
- https://github.com/damap-org/damap-frontend/pull/492/files

ValentinFutterer reviewed (approved):
👍
